### PR TITLE
Improved Logger Initialization and Configurability

### PR DESCRIPTION
**Pull request by nikostoulas**
## Description
This pull request includes changes that enhance the logger initialization process and provide the ability to change log levels for different categories.

## Changes
- Fixed and optimized the logger initialization process to delay and cache it.
- Introduced the ability to change log levels for different log categories.

Aims to convert many "initializing.x.y" logger names to just 'x.y'

### CODE REVIEW

File: docs/logs.md [hunk 1]
```diff
@@ -48,3 +55,19 @@ logger.info('msg');
 ```
 
 
+## Changing logger level
```
Comment (nikostoulas) [inline docs/logs.md:58]: @stavros-wb you were recently asking for this!

File: docs/logs.md [hunk 2]
```diff
@@ -48,3 +55,19 @@ logger.info('msg');
...
+
+
+By updating the config.log.categories you can affect the log level of a specific log category:
+
+e.g.:
+`LOG_CATEGORIES_ORKA_KAFKA=warn LOG_JSON=false node app.js`
+
+Note that log4js supports changing all orka.* log levels not specifically set to a different level by changing 
+the okra level.
```
Comment (stavros-wb) [inline docs/logs.md:67]:
```suggestion
the orka level. 
```

File: docs/logs.md [hunk 3]
```diff
@@ -48,3 +55,19 @@ logger.info('msg');
...
+
+By updating the config.log.categories you can affect the log level of a specific log category:
+
+e.g.:
+`LOG_CATEGORIES_ORKA_KAFKA=warn LOG_JSON=false node app.js`
+
+Note that log4js supports changing all orka.* log levels not specifically set to a different level by changing 
+the okra level. 
+Changing the orka level will not change the orka.kafka.consumer and orka.kafka.producer level as those levels are 
+specifically set to info by default. You need to overwrite those values too in order to change them.
```
Comment (stavros-wb) [inline docs/logs.md:69]: 👍

**Review by stavros-wb — APPROVED** [review]

File: docs/configuration.md [hunk 4]
```diff
@@ -162,7 +162,14 @@ You can find the default values below:
...
     "console": true,
-    "json": false
+    "json": false,
+    "categories": {
+      "orka.kafka.consumer": "info",
+      "orka.kafka.producer": "info",
+      "orka": "debug",
+      "kafka": "debug",
+      "initializing": "debug"
+    }
```
Comment (manoskouvarakis) [inline docs/configuration.md:172]: 👏

**Review by manoskouvarakis — APPROVED** [review]

**Pull request merged**